src/data/num/prime.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2014 Microsoft Corporation. All rights reserved.
```

Comment:
Is this really from 2014 and under Microsoft copyright?

Either way, do you mind adding a module doc for this file?